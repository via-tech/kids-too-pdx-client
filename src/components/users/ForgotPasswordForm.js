import React from 'react';
import PropTypes from 'prop-types';
import { withError } from '../wrappers/withError';

function ForgotPassword({ handleChange, handleSubmit, user, message }) {
  return (
    <main>
      <form onSubmit={handleSubmit.bind(null, user)}>
        <p>Enter the username or email associated with the account</p>
        <label>Username/Email
          <input type="text" name="username" onChange={handleChange} required />
        </label>

        <button type="submit">Recover</button>
      </form>

      {message && <p>{message}</p>}
    </main>
  );
}

ForgotPassword.propTypes = {
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  message: PropTypes.string,
  user: PropTypes.object
};

export default withError(ForgotPassword);
