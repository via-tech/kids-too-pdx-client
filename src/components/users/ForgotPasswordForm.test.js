import React from 'react';
import renderer from 'react-test-renderer';
import ForgotPasswordForm from './ForgotPasswordForm';

describe('Forgot Password component', () => {
  it('snapshots ForgotPassword', () => {
    const props = {
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      message: ''
    };

    const tree = renderer.create(
      <ForgotPasswordForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots component with message', () => {
    const props = {
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      message: 'Temporary password has been sent to test@email.com'
    };

    const tree = renderer.create(
      <ForgotPasswordForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
