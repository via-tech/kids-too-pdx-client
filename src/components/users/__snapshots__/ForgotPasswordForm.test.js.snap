// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password component snapshots ForgotPassword 1`] = `
<div
  className="Wrappers"
>
  <main>
    <form
      onSubmit={[Function]}
    >
      <p>
        Enter the username or email associated with the account
      </p>
      <label>
        Username/Email
        <input
          name="username"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <button
        type="submit"
      >
        Recover
      </button>
    </form>
    
  </main>
</div>
`;

exports[`Forgot Password component snapshots component with message 1`] = `
<div
  className="Wrappers"
>
  <main>
    <form
      onSubmit={[Function]}
    >
      <p>
        Enter the username or email associated with the account
      </p>
      <label>
        Username/Email
        <input
          name="username"
          onChange={[MockFunction]}
          required={true}
          type="text"
        />
      </label>
      <button
        type="submit"
      >
        Recover
      </button>
    </form>
    <p>
      Temporary password has been sent to test@email.com
    </p>
  </main>
</div>
`;
